### PR TITLE
MNT: Python 3.12 support, drop Python 3.7 and pkg_resources

### PR DESCRIPTION
**Pull request by effigies**
From niworkflows tests, it looks like setuptools has finally killed off pkg_resources. Need to get a templateflow release without it.

### CODE REVIEW

**Comment by effigies** [comment]
@mgxd Could I trouble you for a review?

File: templateflow/_loader.py
Comment (mgxd) [inline templateflow/_loader.py:1]: No need to do this - but this feels useful / general enough to warrant standalone packaging

**Review by mgxd — APPROVED** [review]
LGTM, and +1 for a much needed new release!

File: templateflow/_loader.py
Comment (effigies) [reply templateflow/_loader.py:1]: Yeah. I'm starting to think that. On the other hand, I'm not sure how much I love the idea of an extra dependency. I'm starting to think about doing something like setuptools_scm and making a build-time tool that will vendor itself to a desired location.

**Pull request merged**